File: benchmarks/compare/raptor/python/raptor/time_parser.py
"""Parses "HH:MM:SS" time strings to seconds from midnight, cached.

Mirrors src/gtfs/TimeParser.ts.
"""
from __future__ import annotations

from typing import Dict


class TimeParser:
    def __init__(self) -> None:
        self.timeCache: Dict[str, int] = {}

    def getTime(self, time: str) -> int:
        if time not in self.timeCache:
            hh, mm, ss = time.split(":")
            self.timeCache[time] = int(hh) * 60 * 60 + int(mm) * 60 + int(ss)

        return self.timeCache[time]

File: benchmarks/compare/raptor/node/src/gtfs/TimeParser.js
/**
 * Parses time strings and returns them as seconds from midnight. Caches results.
 */
export class TimeParser {
  constructor() {
    this.timeCache = {};
  }

  /**
   * Convert a time string to seconds from midnight
   */
  getTime(time) {
    if (!Object.hasOwn(this.timeCache, time)) {
      const [hh, mm, ss] = time.split(":");

      this.timeCache[time] = (+hh) * 60 * 60 + (+mm) * 60 + (+ss);
    }

    return this.timeCache[time];
  }
}
